src/app/components/sidenav/sidenav.component.html

```diff
@@ -17,7 +17,7 @@
                 </ng-container>
                 <ng-container *ngIf="userRole==roleType.Admin">
                     <ng-container  *ngFor="let item of primaryMenuItemsAdmin">
-                        <button mat-button class="icon-description">
+                        <button mat-button class="icon-description" (click)="item.title==='Categories'? viewCategories() : null">
```

Comment:
Try to not use strings in html, such as 'Categories' is used here. It would be awesome to have a class of Constants and to take this values from there. As a tip, use capslock for constants. So your code would look like item.title === constants.CATEGORIES, or something like that